### DISCUSSION TITLE
How does plugin README get parsed for TWIN?

### DISCUSSION THREAD

**Discussion by mikesmithgh** [codicocodes/dotfyle#70] [Q&A] [answered]
How do I know what data will appear in TWIN after adding a new plug-in to https://github.com/rockerBOO/awesome-neovim? (i.e., screenshot, short description, long description)

Is there a preferred format of the README to make sure it is parsed correctly?

Thanks!

**Comment by codicocodes** [comment, marked as answer]
Hi @mikesmithgh

# Images
Yes there is a preferred format for parsing images. The preferred way is to provide absolute paths to the image in your Readme, Dotfyle can currently not parse images with relative paths. It should be a permanent url and not a url that would dissapear if you delete the image. It also only supports images hosted on GitHub.

I sometimes have the time to manually go through Readme's and try to get images when they cannot be automatically parsed but this is quite time consuming.

One way to get permanent image urls is to upload the image to an issue, I commonly use this issue for when I need to create permaurls: https://github.com/codicocodes/dotfyle/issues/43

This was the preferred way for TWiN in the past as well.

# Text
The text that's parsed is automatically is currently only the repository description. 

I sometimes also go through repositories and manually look for a good long description. If the beginning of the Readme has a self contained paragraph in the third person listing features or why you would use the plugin I often copy it over the This Week in Neovim.

I might consider some way to automatically parse this, but unsure if it'd work well.

# Future

I want to build features to further improve twin:
- Allow community members to add plugins to Dotfyle
- Allow community members to upload updates or tips that can be used in TWiN

**Reply by mikesmithgh** [reply]
@codicocodes Thanks for the info! Does it just parse the first image in the README?

For example, I recently released https://github.com/mikesmithgh/kitty-scrollback.nvim. Due to the formatting of the README, the logo appears first but it would have been preferred if the demo was on TWIN. I am assuming it just grabs the first parsable image url and uses that.

**Reply by codicocodes** [reply]
I don't think I select any particular ordering for the images, so it'd be just what postgres returns first: https://github.com/codicocodes/dotfyle/blob/main/src/lib/server/twin/builder.ts#L84

I believe that's usually the first image in the Readme, but I am not sure I actually specify an ordering. Even if I did specify order I'm not sure the first image in the Readme is always created first.

Most of TWiN is still manually curated and edited, so if I don't think the image is good I look into changing it, I did like your image because it was unique! Although I totally understand the second one could have been preferable in hindsight but I didn't think about it at the time..

**Reply by codicocodes** [reply]
Additionally, I believe my markdown parser strips videos, so currently it accepts only images and gifs, not videos.

**Reply by codicocodes** [reply]
Updated the image for kitty scrollback in the most recent twin!

**Reply by mikesmithgh** [reply]
Awesome, thanks for the details and updating the image!
